core/templates/pages/exploration-player-page/modals/display-hint-modal.component.ts

```diff
@@ -33,12 +34,12 @@ import { PlayerTranscriptService } from '../services/player-transcript.service';
   templateUrl: './display-hint-modal.component.html'
 })
 export class DisplayHintModalComponent {
-  index: number;
-  COMPONENT_NAME_HINT: string;
-  hint: SubtitledHtml;
-  displayedCard: StateCard;
-  recordedVoiceovers: RecordedVoiceovers;
-  hintContentId: string;
+  index: number | undefined;
```

Comment:
Why is `undefined` allowed here?

Reply:
Actually, `index` has been initialised in class `DisplayHintModalComponent`, and in the `ngOnInit()` function, value of `index` has not been set. So it remains undefined here.

Reply:
Can you use `!` then?

Reply:
Done!